ocaml/xapi/message_forwarding.ml

```diff
@@ -3921,14 +3921,22 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 		let attach ~__context ~self =
 			info "Pool_update.attach: pool update = '%s'" (pool_update_uuid ~__context self);
 			let local_fn = Local.Pool_update.attach ~self in
-			VDI.forward_vdi_op ~local_fn ~__context ~self:(Db.Pool_update.get_vdi ~__context ~self)
+			let update_vdi = Db.Pool_update.get_vdi ~__context ~self in
+			if Db.is_valid_ref __context update_vdi then
+			begin VDI.forward_vdi_op ~local_fn ~__context ~self:update_vdi
```

Comment:
We don't need the `begin` and `end` here, but I don't see any harm in keeping them.